gulp-tasks/build-packages.js

```diff
@@ -6,17 +6,33 @@
   https://opensource.org/licenses/MIT.
 */
 
-const fse = require('fs-extra');
+const del = require('del');
+const fs = require('fs-extra');
 const gulp = require('gulp');
 const path = require('path');
 
 const constants = require('./utils/constants');
 const packageRunnner = require('./utils/package-runner');
 
-const cleanPackage = (packagePath) => {
-  const outputDirectory = path.join(packagePath,
-      constants.PACKAGE_BUILD_DIRNAME);
-  return fse.remove(outputDirectory);
+const cleanPackage = async (packagePath) => {
+  // Delete generate files from the the TypeScript transpile.
+  if (await fs.pathExists(path.join(packagePath, 'src', 'index.ts'))) {
+    const deletedPaths = await del([
+      path.join(packagePath, '**/*.+(mjs|d.ts)'),
+      // Don't delete files in node_modules
+      '!**/node_modules', '!**/node_modules/**/*',
+    ]);
+
+    const directoriesToDelete = new Set();
```

Comment:
I don't understand what this bit of code is deleting.

Reply:
When TypeScript generates `.mjs` and `.d.ts` files, it does so keeping the same directory structure as was in `/src`, so any sub-directories will get copied to the root-level package directory, and this code deletes those.

I've added a comment to help better explain what this does.